### PR TITLE
Dg final edits - condensing tag, filter

### PR DESCRIPTION
**Pull request by whoisjustinngo**
- [x] condense tag 
- [x] condense filter
- [x] condense add

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -284,66 +284,60 @@ show feedback to user.
...
 
-### Filtering Persons
+### Filtering Data
 
 #### Implementation
 
-To allow the user to filter based on any attributes of a `Person`, a `Predicate` class is created
-for each attribute. The `ModelManager` manages all data, which allows us to easily filter `Person`s based
-on their atrributes by passing in the correspionding `Predicate`. 
+The data on the `schedule`, `contacts` and `todos`  tabs can be filtered. The underlying implmentation of the `filter` commnds on each of these tabs are the same, and involves `Predicate`s that looks for entries that match the keyword(s) specified by the user.
```
Comment (kslui99) [inline docs/DeveloperGuide.md:291]: 1. Extra whitespace between "todos" and "tabs"
2. commnds typo
3. Tabs: standardise to Schedule, Contacts and Todos tabs, without enclosing in code blocks

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -284,66 +284,60 @@ show feedback to user.
...
 
-The proposed feature of scheduling is parsed by `AddScheduleCommandParser`, and executed by the `AddScheduleCommand`, where it will add the new Event into the list of Events in the users schedule. While adding a new `Event` into the `Schedule`, it will help to check if the given `Event` clashes with the Events which are already in the `Schedule`. 
+As such, each `Predicate` is meant to apply a filter a particular field of a specific `data type`, and keep only the entries that have the field matching the `keyword(s)` specified by the user. After the user has specified the field and keyword(s) to filter by, a corresponding `Predicate` will be passed as an argument into the `Model#updateFiltered<data type>List()` methods for the corresponding `data type`, which in turn makes use of the built-in `setPredicate(Predicate)` method of the javafx `FilteredList<data type>` class, which is the wrapper class for the `ObservableList<data type>` which all the data types use to store all their entries. This method filters the list based on the predicate supplied, and the resulting filtered list with only the entries that the user wants is displayed to the user.
```
Comment (kslui99) [inline docs/DeveloperGuide.md:297]: apply a filter *to* a particular field

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -284,66 +284,60 @@ show feedback to user.
...
 
-### Finding and Editing existing Events
+### Adding Tags to Entries
 
 #### Implementation
 
-The proposed feature of finding and editing existing `Events` is parsed by `FindScheduleCommandParser` and `EditScheduleCommandParser` respectively. Once parsed, those command will be executed by the `FindScheduleCommand` and `EditScheduleCommand` respectively. 
+Tags can be added to the entries on the `schedule`, `contacts` and `todos`  tabs.
+
+These tags have lists that hold `Schedule`, `Person` and `Todo` classes respectively. Let us refer to these as `data type`s. Each of these has a model that encapsulates information for multiple fields. For example, the `Person` model has the fields `Name`, `Relationship`, `Email` etc. that holds the information for each `Person` entry. In addition to these fields, each of these models also has a field called `tags` that tracks the tags that have been applied to that entry.
```
Comment (kslui99) [inline docs/DeveloperGuide.md:313]: These tags have lists that hold `Schedule`, `Person` and `Todo` classes respectively.

Wait wait I don't think this is applicable for Todos. I'm not sure if the others did this though.
Tags is just an attribute of Todo.

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (whoisjustinngo) [reply docs/DeveloperGuide.md:291]: changed

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (whoisjustinngo) [reply docs/DeveloperGuide.md:297]: changed

File: docs/DeveloperGuide.md (hunk 3 above)
Comment (whoisjustinngo) [reply docs/DeveloperGuide.md:313]: bruh my bad its a typo, its supposed to be these tabs

**Pull request merged**